doc/source/user_guide/sparse.rst

```diff
@@ -35,21 +36,64 @@ large, mostly NA ``DataFrame``:
 
    df = pd.DataFrame(np.random.randn(10000, 4))
    df.iloc[:9998] = np.nan
-   sdf = df.to_sparse()
+   sdf = df.astype(pd.SparseDtype("float", np.nan))
    sdf
-   sdf.density
+   sdf.sparse.density
 
 As you can see, the density (% of values that have not been "compressed") is
 extremely low. This sparse object takes up much less memory on disk (pickled)
 and in the Python interpreter. Functionally, their behavior should be nearly
 identical to their dense counterparts.
 
-Any sparse object can be converted back to the standard dense form by calling
-``to_dense``:
+.. _sparse.array:
+
+SparseArray
+-----------
+
+:class:`SparseArray` is a :class:`~pandas.api.extensions.ExtensionArray`
+for storing an array of sparse values (see :ref:`basics.dtypes` for more
+on extension arrays). It is a 1-dimensional ndarray-like object storing
+only values distinct from the ``fill_value``:
+
+.. ipython:: python
+
+   arr = np.random.randn(10)
+   arr[2:5] = np.nan
+   arr[7:8] = np.nan
+   sparr = pd.SparseArray(arr)
+   sparr
+
+A sparse array can be converted to a regular (dense) ndarray with :meth:`numpy.asarray`
```

Comment:
While reviewing, I opened a PR to deprecate `.values`: https://github.com/pandas-dev/pandas/pull/26421 (yet another and confusing way to get the *dense* values and not the sparse values)